### ISSUE TITLE
Field measures not strengthened when exported

### ISSUE THREAD

**Issue by ranjitjhala** [ucsd-progsys/liquidhaskell#295]
Why does tests/todo/measfield.hs not work? 

(Also: it works fine if I replace the line at the top to: `module Goo ()` i.e. don't export `Vec` or any constructors.

1. It shouldn't make any difference whether the type is exported or not?
2. The current set up makes using field measures useless as I have to write a separate accessor...

**Comment by nikivazou** [comment]
You are right, now we do not strengthen the haskell's record selector `velems` at all.
You do not have to write a separate selector, you can refine the existing one (until liquidHaskell automatically refines it)

For your example to work you have to add 

~~~
{-@ velems :: x:Vec a -> {v: [a] | v = velems x && len v = vsz x} @-}
~~~

Note that `velems` is an internal haskell function (something like `velems x = case x of V _ y -> y`) and the above type *is* checked.


If you export `Vec` you also export `velems` and the function `velems` gets refined with true.
If you do not export it,  `velems` gets a refined signature that is appropriate solved. 
Again, `velems` is treated as a usual haskell function.

**Comment by ranjitjhala** [comment]
Hmm. I see, the key bit is that its treated
just like a Haskell function, and really the
"automatic" strengthening is due to our intra
module inference.

This needs to be pointed out explicitly.

Can you add a line to the README.md?

**Comment by gridaphobe** [comment]
I think I've pointed this out a few times (there's probably even an open issue somewhere). I'd rather just fix it than add a note to the README, shouldn't be too hard right?

**Comment by ranjitjhala** [comment]
Is this easy to add? If so, lets just do it because its very confusing otherwise.
